packages/api/src/external/carequality/process-outbound-patient-discovery-resps.ts

```diff
@@ -102,6 +136,129 @@ function buildCQLinks(pdResults: OutboundPatientDiscoveryResp[]): CQLink[] {
       oid: pd.gateway.oid,
       url,
       id: pd.gateway.id,
+      ...(pd.patientMatch ? { patientResource: pd.patientResource } : undefined),
     };
   });
 }
+
+export async function runNextPdOnNewDemographics({
+  patient,
+  facilityId,
+  requestId,
+  cqLinks,
+}: {
+  patient: Patient;
+  facilityId: string;
+  requestId: string;
+  cqLinks: CQLink[];
+}): Promise<boolean> {
+  const updatedPatient = await getPatientOrFail(patient);
+
+  const newDemographicsHere = getNewDemographics(updatedPatient, cqLinks);
+  const foundNewDemographicsHere = newDemographicsHere.length > 0;
+  const foundNewDemographicsAcrossHies = await checkLinkDemographicsAcrossHies({
+    patient: updatedPatient,
+    requestId,
+  });
+  if (foundNewDemographicsHere || foundNewDemographicsAcrossHies) {
+    if (foundNewDemographicsHere) {
+      await updatePatientLinkDemographics({
+        requestId,
+        patient: updatedPatient,
+        source: MedicalDataSource.CAREQUALITY,
+        links: newDemographicsHere,
+      });
+    }
+    await discover({
+      patient: updatedPatient,
+      facilityId,
+      rerunPdOnNewDemographics: false,
+    });
+    analytics({
+      distinctId: updatedPatient.cxId,
+      event: EventTypes.patientDiscovery,
+      properties: {
+        hie: MedicalDataSource.CAREQUALITY,
+        patientId: updatedPatient.id,
+        requestId,
+        foundNewDemographicsHere,
+        foundNewDemographicsAcrossHies,
+      },
+    });
+    return true;
+  }
+  return false;
+}
+
+export async function runNexPdIfScheduled({
+  patient,
+  requestId,
+}: {
+  patient: Patient;
+  requestId: string;
+}): Promise<boolean> {
+  const updatedPatient = await getPatientOrFail(patient);
+
+  const scheduledPdRequest = getCQData(updatedPatient.data.externalData)?.scheduledPdRequest;
+
+  if (scheduledPdRequest) {
+    await discover({
+      patient: updatedPatient,
+      facilityId: scheduledPdRequest.facilityId,
+      requestId: scheduledPdRequest.requestId,
+      forceEnabled: scheduledPdRequest.forceCarequality,
+      rerunPdOnNewDemographics: scheduledPdRequest.rerunPdOnNewDemographics,
+    });
+    await resetPatientScheduledPatientDiscoveryRequestId({
+      patient: updatedPatient,
+      source: MedicalDataSource.CAREQUALITY,
+    });
+    analytics({
+      distinctId: updatedPatient.cxId,
+      event: EventTypes.runScheduledPatientDiscovery,
```

Comment:
I think we can just call this `patientAugmentation` as well. I dont think its important whether or no its scheduled (in the naming). We can add as a property that its scheduled

Reply:
I think this is just to flag if we ended up running a scheduled PD. Just want to see how frequently (if ever) the scheduling framework gets used.